lib/Doctrine/ORM/Mapping/DefaultTypedFieldMapper.php

```diff
@@ -36,12 +57,47 @@ final class DefaultTypedFieldMapper implements TypedFieldMapper
         'string' => Types::STRING,
     ];
 
+    /** It maps built-in Doctrine types to PHP types */
+    private const BUILTIN_TYPES_MAP = [
+        AsciiStringType::class => 'string',
+        BigIntType::class => 'string',
+        BooleanType::class => 'bool',
+        DateType::class => DateTime::class,
+        DateImmutableType::class => DateTimeImmutable::class,
+        DateIntervalType::class => DateInterval::class,
+        DateTimeType::class => DateTime::class,
+        DateTimeImmutableType::class => DateTimeImmutable::class,
+        DateTimeTzType::class => DateTime::class,
+        DateTimeTzImmutableType::class => DateTimeImmutable::class,
```

Comment:
Careful: in these cases, `DateTimeInterface` is perfectly valid!

I suggest skipping on any types where objects are de-serialized